File: src/cpp/Longest-Substring-Without-Repeating-Characters.cpp
class Solution
{
    public:
        int lengthOfLongestSubstring(string s)
        {
            unordered_map<char, int> m;
            int i = 0;
            int j = 0;
            int ans = 0;
            while (j < s.size())
            {
                m[s[j]]++;
                if (j - i + 1 == m.size())
                {
                    ans = max(ans, j - i + 1);
                }
                else
                {
                    while (m.size() < j - i + 1)
                    {
                        m[s[i]]--;
                        if (m[s[i]] == 0)
                        {
                            m.erase(s[i]);
                        }
                        i++;
                    }
                    ans = max(ans, j - i + 1);
                }
                j++;
            }
            return ans;
        }
};

File: src/java/longest-substring-without-repeating-characters.java
/*Question:-Given a string s, find the length of the longest substring without repeating characters.

 

Example 1:

Input: s = "abcabcbb"
Output: 3
Explanation: The answer is "abc", with the length of 3.*/

//Solution

import java.util.*;
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int left = 0;
        int right = 0;
        int length = s.length();
        int max_length = 0;
        Set<Character> check = new HashSet<>();
        
        while(right < length){
            if(check.contains( s.charAt(right)) == true){
                
                while((left < right) && check.contains(s.charAt(right)) == true){
                    check.remove(s.charAt(left));
                    left++;
                }
            }
            
            check.add(s.charAt(right));
            max_length = Math.max(max_length, right-left+1);
            right++;
            
        }
        
        return max_length;
    }
}

//Submission link:-https://leetcode.com/submissions/detail/797209410/
